src/bridge.js

```diff
@@ -445,7 +445,7 @@ Bridge.prototype._customiseAppserviceThirdPartyLookup = function(lookupControlle
 
         this.addAppServicePath({
             method: "GET",
-            path: "/_matrix/app/:version/thirdparty/protocol/:protocol",
+            path: "/_matrix/app/:version(v1|unstable)/thirdparty/protocol/:protocol",
```

Comment:
`/:version(v1|unstable)/` is the same as `/(v1|unstable)/`?
It does not allow URLs like `/invalidv1/` where `version="invalid"`?

Reply:
Yep, these format restrictions for route params are described over here:
https://expressjs.com/en/guide/routing.html#route-parameters

Reply:
![image](https://user-images.githubusercontent.com/2072976/79461312-e853bc80-7fed-11ea-9c3b-16f01aee7cf8.png)

Try `/_matrix/app/(v1|unstable)/thirdparty/protocol/:protocol`
and `/_matrix/app/:version(v1|unstable)/thirdparty/protocol/:protocol`
against the test string `/_matrix/app/v1/thirdparty/protocol/p`

http://forbeslindesay.github.io/express-route-tester/

Apparently if the group is not named, it doesn't work.

Reply:
> It does not allow URLs like /invalidv1/ where version="invalid"?

Oh and yes, correct. Just like a RegExp